### ISSUE TITLE
Allow for custom Opensearch index configuration to allow HA setups

### ISSUE THREAD

**Issue by mtneug** [opencast/opencast#2136]
**Describe the bug**

The Elasticsearch index configuration is hardcoded as Java resource. It should instead be a configuration file to allow for a HA configuration with multiple replicas.

**Comment by KatrinIhler** [comment]
Is this fixed by https://github.com/opencast/opencast/pull/6491?

**Comment by mtneug** [comment]
No. These two files need to be configurable (at least `number_of_shards` and `number_of_replicas`):

* https://github.com/opencast/opencast/blob/739cf40731ebdde073a48f14a3324d1f3e725f35/modules/elasticsearch-impl/src/main/resources/elasticsearch/indexSettings.json#L2-L3
* https://github.com/opencast/opencast/blob/739cf40731ebdde073a48f14a3324d1f3e725f35/modules/elasticsearch-impl/src/main/resources/elasticsearch/indexSettings.json#L2-L3

**Comment by lkiesow** [comment]
@mtneug, I think this is possible by now. Isn't it? Can this be closed?

**Comment by mtneug** [comment]
> @mtneug, I think this is possible by now. Isn't it? Can this be closed?

How? I don't think the number of shards and replicas is configurable yet.

**Comment by gregorydlogan** [comment]
NB: The search service creates its index in a slightly different way, so if you're building this configurability you should check that separately.  It may work the way the rest do, it may not - I don't remember, and haven't had time to fix that yet.
